[WIP][CS2] Add webpack support

akfish opened the pull request:
Resend #4501

GeoffreyBooth commented on Cakefile:
```diff
@@ -440,3 +435,30 @@ task 'test:browser', 'run the test suite against the merged browser script', ->
+task 'build:webpack', 'build webpack test bundle', ->
+  args = [
+    "./node_modules/webpack/bin/webpack.js"
+    "--entry=./test/webpack/entry.js", 
```
Why this entry, and not browser.js as above?

GeoffreyBooth commented on Cakefile:
```diff
@@ -440,3 +435,30 @@ task 'test:browser', 'run the test suite against the merged browser script', ->
+
+task 'test:webpack', 'run test suite against the webpack test bundle', ->
+  loader = """
+  var __exports__ = {};
```
What is this variable? Is it distinct from CommonJS `exports`?

akfish replied:
Webpack will scope known globals. It is not possible to access them directly. This, along with the custom entry file and the loader script in test:webpack task, allows the bundled CoffeeScript to escape thus it can be required in node environment.

akfish replied:
There might be a better way to do this. I'll get back to you later.

akfish commented:
Turns out there's an webpack option for that and no additional configuration files nor plugins required. That makes more sense. Don't know what I was thinking back then.
I changed the entry point to `./` instead of explicitly specify one. This let webpack to use `browser` field in `package.json`.
I also removed the `build:webpack` task and have the build run in `test:webpack` task.

GeoffreyBooth commented:
@lydell or @jashkenas did you have any further notes? I’m leaving the `test:integrations` task as a place for future tests like loading `test.html` via Selenium Chromedriver.

Somehow `cake test:browser` fails both here and on `2`, so something we did recently broke the browser build; but it’s unrelated to this PR.

jashkenas commented:
Looks great!
